lib/formats/odata.js

```diff
@@ -70,8 +70,14 @@ const nextUrlFor = (limit, offset, count, originalUrl) =>
     : urlWithQueryParams(originalUrl, { $skip: (offset + limit), $top: null }));
 
 // Given a querystring object, returns an object of relevant OData options. Right
-// now that is only { wkt: Bool, expand: String }
-const extractOptions = (query) => ({ wkt: isTrue(query.$wkt), expand: query.$expand });
+// now that is only { wkt: Bool, expand: String, metadata: Array }
+const extractOptions = (query) => ({
+  wkt: isTrue(query.$wkt),
+  expand: query.$expand,
+  metadata: query.$select?.split(',')
+    .map(p => p.trim().toLowerCase())
+    .filter(p => p.startsWith('__'))
+});
```

Comment:
it seems to me most of your use of the `metadata` object is as a lookup, not as a list. can you use a `{}` instead of an array here?

Reply:
Thanks, I have refactored the code